x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/FollowIndexAction.java

```diff
@@ -275,24 +276,43 @@ void finalizeResponse() {
         }
     }
 
-
     static void validate(IndexMetaData leaderIndex, IndexMetaData followIndex, Request request) {
         if (leaderIndex == null) {
             throw new IllegalArgumentException("leader index [" + request.leaderIndex + "] does not exist");
         }
-
         if (followIndex == null) {
             throw new IllegalArgumentException("follow index [" + request.followIndex + "] does not exist");
         }
         if (leaderIndex.getSettings().getAsBoolean(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), false) == false) {
             throw new IllegalArgumentException("leader index [" + request.leaderIndex + "] does not have soft deletes enabled");
         }
-
         if (leaderIndex.getNumberOfShards() != followIndex.getNumberOfShards()) {
             throw new IllegalArgumentException("leader index primary shards [" + leaderIndex.getNumberOfShards() +
                 "] does not match with the number of shards of the follow index [" + followIndex.getNumberOfShards() + "]");
         }
-        // TODO: other validation checks
+        if (leaderIndex.getRoutingNumShards() != followIndex.getRoutingNumShards()) {
+            throw new IllegalArgumentException("leader index number_of_routing_shards [" + leaderIndex.getRoutingNumShards() +
+                "] does not match with the number_of_routing_shards of the follow index [" + followIndex.getRoutingNumShards() + "]");
+        }
+        if (leaderIndex.getState() != IndexMetaData.State.OPEN || followIndex.getState() != IndexMetaData.State.OPEN) {
+            throw new IllegalArgumentException("leader and follow index must be open");
+        }
+        Map<String, Object> leaderMapping = getMapping(leaderIndex);
+        Map<String, Object> followerMapping = getMapping(followIndex);
+        if (leaderMapping.equals(followerMapping) == false) {
```

Comment:
FYI - for future work - they should be mergeable (i.e. leader into follower) not necessarily identical.

Reply:
That makes sense. The shard follow tasks also add the missing fields in the case they are missing at startup.

Reply:
done here: 284e2ddd6f2c6da0d0255d6a8d45d8cd6e9ebddf